File: MoniGuardAPI/Face.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace MoniGuardAPI
{
    public class Face(int faceId, int guestId, string? name, DateTime capturedAt, byte[]? hash, byte[]? content)
    {
        public Face(int faceId, int guestId, string? name, DateTime capturedAt, byte[]? hash) :
            this(faceId,guestId,name,capturedAt,hash,null)
        {

        }

        public Face() : this(0, 0, null, DateTime.Now, null)
        { 
        }


        [Key]
        public int FaceId { get; set; } = faceId;
        [ForeignKey("Guest")]
        public int GuestId { get; set; } = guestId;
        public string? Name { get; set; } = name;
        public DateTime CapturedAt { get; set; } = capturedAt;
        public byte[]? Hash { get; set; } = hash;

        [JsonIgnore]
        public byte[]? Content { get; set; } = content;
    }
}

File: moni_guard_api/face.py
import datetime


class Face:
    def __init__(self, face_id: int, guest_id: int, name: str, captured_at: datetime, hash: bytes):
        """
        :param face_id: 人脸 ID。
        :param guest_id: 访客 ID。
        :param name: 人脸名称。
        :param captured_at: 人脸捕获时间。
        :param hash: 人脸哈希值。
        """
        self.face_id = face_id
        self.guest_id = guest_id
        self.name = name
        self.captured_at = captured_at
        self.hash = hash

    def __str__(self):
        return f"Face(face_id={self.face_id}, face_encoding={self.face_encoding}, face_image={self.face_image})"

    def __repr__(self):
        return str(self)